src/pytorch_lightning/trainer/trainer.py

```diff
@@ -1930,13 +1925,18 @@ def reset_val_dataloader(self, model: Optional["pl.LightningModule"] = None) ->
         has_step = is_overridden("validation_step", pl_module)
         enable_validation = self.limit_val_batches > 0
         if source.is_defined() and has_step and enable_validation:
+            # store epoch of dataloader reset for reload_dataloaders_every_n_epochs
+            # it should not reload again if it has already reloaded during sanity_check
+            if self.state.fn == TrainerFn.FITTING and (
+                (self.sanity_checking and self.fit_loop.epoch_loop._should_check_val_epoch())
+                or not self.sanity_checking
+            ):
+                self._last_val_dl_reload_epoch = self.current_epoch
```

Comment:
if sanity check is not happening, this means this is triggered for validation and we should set this value. if sanity check is happening, we need to check if validation is going to happen in the same epoch or not. If not we should not set this value to avoid the effect of sanity check on `reload_dataloaders_every_n_epoch`, but if it will happen, we don't reload the dataloaders again and set this value to avoid the reload in the same epoch during validation.